### DISCUSSION TITLE
"Conditional" Workflow Labels? (Conditional workflow resource requirements)

### DISCUSSION THREAD

**Discussion by gdevenyi** [nextflow-io/nextflow#4750] [Q&A] [answered]
I have two `process`es which are used in two different sub-workflows of a nextflow main workflow.

While the commands are the same, the resources they need for execution are dependent on the input files they receive, which in turn are dependent on which sub-workflow they execute in. Is there a way to detect which sub-workflow the process is executing under and use that? In particular in this case, I'd like to take advantage of the profiles to change the resource requests for certain clusters via labels of the processes.

**Comment by gdevenyi** [comment]
It looks like this info is available somewhere internally, as individual jobs are named like:
`<NAMED_SUB_WORKFLOW>:<DEEPER_NAMED_SUBWORKFLOW>:<PROCESS>`

**Reply by bentsherman** [reply]
You can do exactly this with `withName` to target processes within specific workflows. It's not well documented at the moment.

**Reply by gdevenyi** [reply]
If I don't explicitly label a process, will it receive an implicit label:
`<NAMED_SUB_WORKFLOW>:<DEEPER_NAMED_SUBWORKFLOW>:<PROCESS>` that I can target using `withName`?

**Reply by bentsherman** [reply, marked as answer]
withName just targets the name of the process, it is not related to labels. Using withName you can target any of (1) the process base name (2) the imported name or (3) the "fully qualified" name which includes all of the subworkflow prefixes.

**Reply by gdevenyi** [reply]
Thanks, that's very helpful.

Can you please confirm that the colon separated name is the internal `fully qualified` name that I can use in combination with the selector expressions
https://www.nextflow.io/docs/latest/config.html#selector-expressions

**Reply by bentsherman** [reply]
Yes exactly
